docs/agents/database-security.md

```
@@ -1,79 +1,3 @@
-
-// CORRECT: Caller builds SQL fragment
-const filter = sql`eval_id = ${evalId} AND success = ${1}`;
-const results = await queryWithFilter(filter);
-```
-
-## Key Files with Database Queries
-
-- `src/models/eval.ts` - Main eval queries, filter building
-- `src/util/calculateFilteredMetrics.ts` - Metrics aggregation queries
-- `src/database/index.ts` - Database connection
+This project uses Drizzle ORM with SQLite, and all database queries must strictly use parameterized queries to prevent SQL injection attacks and ensure safe database interactions. Developers should always rely on Drizzle's `sql` tagged template literals, as they automatically handle value escaping and significantly reduce the risk of malicious input affecting query execution, for example: `const query = sql\`SELECT * FROM eval_results WHERE eval_id = \${evalId} AND success = \${1}\`;`. It is critical to avoid constructing queries using raw SQL methods such as `sql.raw()` or string interpolation with dynamic values, since these practices can introduce serious security vulnerabilities and allow attackers to manipulate database queries, potentially leading to data leaks, unauthorized access, or data corruption. In certain edge cases like SQLite's `json_extract()` function, where literal string paths are required and cannot be parameterized, any user-provided input must be carefully sanitized and escaped before use, typically by replacing special characters such as backslashes and quotation marks to prevent injection risks. Following these secure coding practices ensures that all database operations remain reliable, predictable, and resistant to common attack vectors, while maintaining the overall integrity and security of the application.
```

markdownlint-cli2 (0.22.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)